### PR TITLE
Issue #4481: Improve readability of suggestions in link autocomplete dropdown

### LINKED ISSUE #4481: [UX] Link autocomplete: Improve the readability/scannability of suggestions in the dropdown
Makes it hard to distinguish the actual options with all the additional info:

Screen Shot 2020-07-14 at 12 38 25 pm

### PR DESCRIPTION
**Pull request by klonos**
Fixes https://github.com/backdrop/backdrop-issues/issues/4481

### CODE REVIEW

File: core/themes/seven/css/style.css [hunk 1]
```diff
@@ -803,7 +803,13 @@ form.views-exposed-form {
 #autocomplete li {
   padding: 0 5px;
 }
-
+#autocomplete li .suggestion {
+  /*font-weight: bold;*/
```
Comment (klonos) [inline core/themes/seven/css/style.css]: Leaving this here for now, till we decide on a style.

File: core/themes/seven/css/style.css [hunk 2]
```diff
@@ -803,7 +803,13 @@ form.views-exposed-form {
 #autocomplete li {
   padding: 0 5px;
 }
-
+.autocomplete-description {
```
Comment (klonos) [inline core/themes/seven/css/style.css:806]: I know that we've discussed reducing this to just the specific class name, but in order to fix the gray in the selected/hovered list items, I had to add the following `#autocomplete li.selected .autocomplete-description` selector, so this now seems kinda odd/silly. Should I change it back to the "full" version?:

```suggestion
#autocomplete li .autocomplete-description {
```

File: core/modules/system/system.module [hunk 3]
```diff
@@ -4056,10 +4056,9 @@ function path_autocomplete($string = '') {
       ->execute();
     foreach ($results as $result) {
       $path = backdrop_get_path_alias('node/' . $result->nid);
-      $matches[$path] = t('@title (Content: %type)', array(
-        '@title' => $result->title,
-        '%type' => node_type_get_name($result->type),
-      ));
+      $match = '<span class="autocomplete-suggestion">' . $result->title . '</span>';
+      $match_description = t('Content: %type', array('%type' => node_type_get_name($result->type)));
+      $matches[$path] = $match . " " . '<span class="autocomplete-description">(' . $match_description . ')</span>';
```
Comment (ghost) [inline core/modules/system/system.module]:
```suggestion
      $matches[$path] = $match . ' <span class="autocomplete-description">(' . $match_description . ')</span>';
```

File: core/modules/system/system.module [hunk 4]
```diff
@@ -4078,10 +4077,9 @@ function path_autocomplete($string = '') {
         $absolute_path = parse_url($GLOBALS['base_url'], PHP_URL_PATH) . '/';
         $url = file_create_url($result->uri);
         $path = str_replace($GLOBALS['base_url'] . '/', $absolute_path, $url);
-        $matches[$path] = t('@filename (File: %type)', array(
-          '@filename' => $result->filename,
-          '%type' => file_type_get_name($result->type),
-        ));
+        $match = '<span class="autocomplete-suggestion">' . $result->filename . '</span>';
+        $match_description = t('File: %type', array('%type' => file_type_get_name($result->type)));
+        $matches[$path] = $match . " " . '<span class="autocomplete-description">(' . $match_description . ')</span>';
```
Comment (ghost) [inline core/modules/system/system.module]:
```suggestion
        $matches[$path] = $match . ' <span class="autocomplete-description">(' . $match_description . ')</span>';
```

File: core/modules/system/system.module [hunk 5]
```diff
@@ -4103,9 +4101,9 @@ function path_autocomplete($string = '') {
         $users = user_load_multiple($uids);
         foreach ($users as $user) {
           $path = backdrop_get_path_alias('user/' . $user->uid);
-          $matches[$path] = t('@name (User account)', array(
-            '@name' => user_format_name($user),
-          ));
+          $match = '<span class="autocomplete-suggestion">' . user_format_name($user) . '</span>';
+          $match_description = t('User account');
+          $matches[$path] = $match . " " . '<span class="autocomplete-description">(' . $match_description . ')</span>';
```
Comment (ghost) [inline core/modules/system/system.module]:
```suggestion
          $matches[$path] = $match . ' <span class="autocomplete-description">(' . $match_description . ')</span>';
```

File: core/modules/system/system.module [hunk 6]
```diff
@@ -4122,10 +4120,9 @@ function path_autocomplete($string = '') {
         ->execute();
       foreach ($results as $result) {
         $path = backdrop_get_path_alias('taxonomy/term/' . $result->tid);
-        $matches[$path] = t('@name (Taxonomy term: %vocabulary)', array(
-          '@name' => $result->name,
-          '%vocabulary' => taxonomy_vocabulary_load($result->vocabulary)->name,
-        ));
+        $match = '<span class="autocomplete-suggestion">' . $result->name . '</span>';
+        $match_description = t('Taxonomy term: %vocabulary', array('%vocabulary' => taxonomy_vocabulary_load($result->vocabulary)->name));
+        $matches[$path] = $match . " " . '<span class="autocomplete-description">(' . $match_description . ')</span>';
```
Comment (ghost) [inline core/modules/system/system.module]:
```suggestion
        $matches[$path] = $match . ' <span class="autocomplete-description">(' . $match_description . ')</span>';
```

File: core/modules/system/system.module [hunk 7]
```diff
@@ -4155,10 +4152,9 @@ function path_autocomplete($string = '') {
                 $view_title = $view->human_name;
               }
               if ($found_match) {
-                $matches[$path] = t('@title (View: %display)', array(
-                  '@title' => $view_title,
-                  '%display' => $view->human_name . ' - ' . $display->display_title,
-                ));
+                $match = '<span class="autocomplete-suggestion">' . $view_title . '</span>';
+                $match_description = t('View: %display', array('%display' => $view->human_name . ' - ' . $display->display_title));
```
Comment (ghost) [inline core/modules/system/system.module]:
```suggestion
                $match_description = t('View: %human - %display', array('%human' => $view->human_name, '%display' => $display->display_title));
```

File: core/modules/system/system.module [hunk 8]
```diff
@@ -4155,10 +4152,9 @@ function path_autocomplete($string = '') {
...
+                $matches[$path] = $match . " " . '<span class="autocomplete-description">(' . $match_description . ')</span>';
```
Comment (ghost) [inline core/modules/system/system.module]:
```suggestion
                $matches[$path] = $match . ' <span class="autocomplete-description">(' . $match_description . ')</span>';
```

File: core/modules/system/system.module [hunk 9]
```diff
@@ -4185,10 +4181,9 @@ function path_autocomplete($string = '') {
             $layout_title = $menu_item['name'];
           }
           if ($found_match) {
-            $matches[$menu_item['path']] = t('@title (Layout: %display)', array(
-              '@title' => $layout_title,
-              '%display' => $menu_item['name'],
-            ));
+            $match = '<span class="autocomplete-suggestion">' . $layout_title . '</span>';
+            $match_description = t('Layout: %display', array('%display' => $menu_item['name']));
+            $matches[$menu_item['path']] = $match . " " . '<span class="autocomplete-description">(' . $match_description . ')</span>';
```
Comment (ghost) [inline core/modules/system/system.module]:
```suggestion
            $matches[$menu_item['path']] = $match . ' <span class="autocomplete-description">(' . $match_description . ')</span>';
```

File: core/modules/system/system.module [hunk 10]
```diff
@@ -4206,10 +4201,9 @@ function path_autocomplete($string = '') {
       foreach ($results as $result) {
         $path = backdrop_get_path_alias($result->link_path);
         if (!isset($matches[$path]) && !path_is_admin($path)) {
-          $matches[$path] = t('@title (Menu item: %path)', array(
-            '@title' => $result->link_title,
-            '%path' => $path,
-          ));
+          $match = '<span class="autocomplete-suggestion">' . $result->link_title . '</span>';
+          $match_description = t('Menu item: %path', array('%path' => $path));
+          $matches[$path] = $match . " " . '<span class="autocomplete-description">(' . $match_description . ')</span>';
```
Comment (ghost) [inline core/modules/system/system.module]:
```suggestion
          $matches[$path] = $match . ' <span class="autocomplete-description">(' . $match_description . ')</span>';
```

File: core/themes/seven/css/style.css [hunk 11]
```diff
@@ -803,7 +803,16 @@ form.views-exposed-form {
...
+  color: #666;
+  font-size: 80%;
+}
+#autocomplete li.selected .autocomplete-description {
+  color: #ddd;
+}
+[dir="rtl"] .autocomplete-suggestion {
+  float: right;
```
Comment (ghost) [inline core/themes/seven/css/style.css]:
```suggestion
  float: right;
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: core/modules/system/system.module (hunk 3 above)
Comment (klonos) [reply core/modules/system/system.module]: Thanks 👍 ...although this is not specifically mentioned in https://api.backdropcms.org/php-standards#string-concatenation, I've fixed it with [`a30ebb0` (#3189)](https://github.com/backdrop/backdrop/pull/3189/commits/a30ebb0d06f134132143d111fc2e1b3ae2a252ec)

File: core/modules/system/system.module (hunk 4 above)
Comment (klonos) [reply core/modules/system/system.module]: Fixed with [`a30ebb0` (#3189)](https://github.com/backdrop/backdrop/pull/3189/commits/a30ebb0d06f134132143d111fc2e1b3ae2a252ec)

File: core/modules/system/system.module (hunk 5 above)
Comment (klonos) [reply core/modules/system/system.module]: Fixed with [`a30ebb0` (#3189)](https://github.com/backdrop/backdrop/pull/3189/commits/a30ebb0d06f134132143d111fc2e1b3ae2a252ec)

File: core/modules/system/system.module (hunk 6 above)
Comment (klonos) [reply core/modules/system/system.module]: Fixed with [`a30ebb0` (#3189)](https://github.com/backdrop/backdrop/pull/3189/commits/a30ebb0d06f134132143d111fc2e1b3ae2a252ec)

File: core/modules/system/system.module (hunk 8 above)
Comment (klonos) [reply core/modules/system/system.module]: Fixed with [`a30ebb0` (#3189)](https://github.com/backdrop/backdrop/pull/3189/commits/a30ebb0d06f134132143d111fc2e1b3ae2a252ec)

File: core/modules/system/system.module (hunk 10 above)
Comment (klonos) [reply core/modules/system/system.module]: Fixed with [`a30ebb0` (#3189)](https://github.com/backdrop/backdrop/pull/3189/commits/a30ebb0d06f134132143d111fc2e1b3ae2a252ec)

File: core/modules/system/system.module (hunk 9 above)
Comment (klonos) [reply core/modules/system/system.module]: Fixed with [`a30ebb0` (#3189)](https://github.com/backdrop/backdrop/pull/3189/commits/a30ebb0d06f134132143d111fc2e1b3ae2a252ec)

File: core/themes/seven/css/style.css (hunk 11 above)
Comment (klonos) [reply core/themes/seven/css/style.css]: Fixed with [`b856ade` (#3189)](https://github.com/backdrop/backdrop/pull/3189/commits/b856adecf090e38f8f1fb3605ffd9d39ab440ae2)

File: core/modules/system/system.module (hunk 7 above)
Comment (klonos) [reply core/modules/system/system.module]: Good catch @BWPanda 👍 ...committed your suggestion with [`3aa4d5b` (#3189)](https://github.com/backdrop/backdrop/pull/3189/commits/3aa4d5ba9238557138302e22fca9e7e5604ba9ff)

**Review by ghost — APPROVED** [review]
Looks good, thanks!

**Comment by klonos** [comment]
...ZenCI acting up

**Comment by quicksketch** [comment]
Merged into 1.x and 1.16.x. Thanks!

**Pull request merged**